kotlin/services/location/README.md

```diff
@@ -0,0 +1,131 @@
+# Amazon Location code examples for the SDK for Kotlin
+
+## Overview
+
+Shows how to use the AWS SDK for Kotlin to work with Amazon Location Service.
+
+<!--custom.overview.start-->
+<!--custom.overview.end-->
+
+_Amazon Location lets you easily and securely add maps, places, routes, geofences, and trackers, to your applications._
+
+## ⚠ Important
+
+* Running this code might result in charges to your AWS account. For more details, see [AWS Pricing](https://aws.amazon.com/pricing/) and [Free Tier](https://aws.amazon.com/free/).
+* Running the tests might result in charges to your AWS account.
+* We recommend that you grant your code least privilege. At most, grant only the minimum permissions required to perform the task. For more information, see [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege).
+* This code is not tested in every AWS Region. For more information, see [AWS Regional Services](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services).
+
+<!--custom.important.start-->
+<!--custom.important.end-->
+
+## Code examples
+
+### Prerequisites
+
+For prerequisites, see the [README](../../README.md#Prerequisites) in the `kotlin` folder.
+
+
+<!--custom.prerequisites.start-->
+<!--custom.prerequisites.end-->
+
+### Get started
+
+- [Hello Amazon Location](src/main/java/com/example/location/HelloLocation.kt#L10) (`ListGeofencesPaginator`)
+
+
+### Basics
+
+Code examples that show you how to perform the essential operations within a service.
+
+- [Learn the basics](src/main/java/com/example/location/scenario/LocationScenario.kt)
+
+
+### Single actions
+
+Code excerpts that show you how to call individual service functions.
+
+- [BatchUpdateDevicePosition](src/main/java/com/example/location/scenario/LocationScenario.kt#L567)
```

Comment:
Why are all of the Kotlin files in a folder named Java?

Reply:
Could be an issue with Kotlin as the folder name - even when Mark as Source Root is specified,